### PR TITLE
Update MinimalLib for Function Exposure: runReactants

### PR DESCRIPTION
**Pull request by syedzayyan**
This is an exposure of runReactants features to minimal lib:

- a single function is exposed
- results are an array of JSMolLists

It passed the simple test I coded. First-time contributor, please go easy :).

### CODE REVIEW

**Comment by syedzayyan** [comment]
Hello! @greglandrum @ptosco 

A silly question. I plan to port ScaffoldNetwork to MinimalLib as well. Would I need to open a new PR for that or shall I merge that into this PR?

**Comment by ptosco** [comment]
@syedzayyan Please submit a separate PR.
Also, I'd recommend to make building ScaffoldNetwork support optional through a CMake switch (mapping to a preprocessor macro, such as `RDK_BUILD_MINIMAL_LIB_SCAFFOLDNETWORK`) in order not to keep MnimalLib as minimal as it can be for non-interested users.

**Comment by syedzayyan** [comment]
Thank you for your prompt response! @ptosco

I'll go ahead and submit a separate PR for the ScaffoldNetwork feature, as suggested with the CMake switch for greater flexibility.

On another note, if you find some time, I'd love to get your input on this PR :).

File: Code/MinimalLib/minilib.cpp [hunk 1]
```diff
@@ -642,9 +641,40 @@ std::string JSReaction::get_svg_with_highlights(
...
+  if (!d_rxn) {
+    std::cerr << "Error: d_rxn is null\n";
+    return {};
+  }
+  d_rxn->initReactantMatchers();
+  RDKit::MOL_SPTR_VECT reactant_vec;
+
+  for (const auto &reactant : reactants.mols()) {
+    if (!reactant) {
+      std::cerr << "Error: reactant is null\n";
```
Comment (greglandrum) [inline Code/MinimalLib/minilib.cpp:657]: We don't do error reporting straight to `std::cerr`. 
I think you need to thrown an exception here, but @ptosco should suggest what he thinks is the best approach for error handling in minilib

**Comment by ptosco** [comment]
@syedzayyan I have submitted a PR against your branch with some suggested changes.
1. You do not need to check `d_rxn` against non-nullness as `JSReaction` cannot be constructed with a null `ChemicalReaction`.
2. @greglandrum is right about the fact that it is better to throw an exception rather than printing anything to `std::cerr`
3. I would not catch upstream exceptions
4. I would return a vector of `JSMolList` pointers and call it `JSMolListList` for consistency with `StringList`
5. I would `delete` all C++ objects in the test; while it is not terrilbly important to leak some memory in the test, I think it is good practice to write tests which cleanly `delete` objects on exit as they may serve as examples to other developers.

**Comment by syedzayyan** [comment]
Hello @ptosco ! Thanks very much for the suggestions. I was wondering why return a vector of `JSMolList` pointers instead, on point 4. The build fails when a pointer is returned and `allow_raw_pointers()` is not helping with the `Implicitly binding raw pointers is illegal` error.

**Review by ptosco — APPROVED** [review]
I tested this locally and it looks good to me.

**Comment by greglandrum** [comment]
I just (hopefully) fixed the merge conflicts. Will merge later this afternoon after the CI builds go through

**Pull request merged**